tests/phpunit/tests/media.php

```diff
@@ -3910,7 +3962,7 @@ static function( $attr ) {
 				'post_content_filtered' => $post_content,
 			)
 		);
-
+		wp_high_priority_element_flag( true );
```

Comment:
`get_the_post_thumbnail` makes a call to `wp_get_attachment_image` which then makes a call to `wp_get_loading_optimization_attributes`.  

As per our early discussion [discussion_r1213463607](https://github.com/WordPress/wordpress-develop/pull/4495#discussion_r1213463607) we have decide to respect what the user has entered and also decided to change the flag if the user manually set an image as high priority.
In case of lazy-loading there is we don't change the count for user inputs. 

This is why we have to reset the flag inbetween `expected` and `output` for fetchpriority.